pandas/core/window/rolling.py

```diff
@@ -498,11 +504,24 @@ def _apply(
 
                 def calc(x):
                     x = np.concatenate((x, additional_nans))
-                    min_periods = calculate_min_periods(
-                        window, self.min_periods, len(x), require_min_periods, floor
-                    )
+                    if not isinstance(window, window_indexers.BaseIndexer):
```

Comment:
when would this not be true?

Reply:
When `df.rolling(2)` or `df.rolling('2s')`, `window` here is still an integer. and the `FixedWindowIndexer` or `Variab leWindowIndexer` respectively is created separately.